### PR TITLE
feat(docs): adding data durability wording

### PR DESCRIPTION
**Pull request by amyshwang**
Capturing https://github.com/questdb/questdb/pull/3304

### CODE REVIEW

File: docs/concept/storage-model.md [hunk 1]
```diff
@@ -73,6 +73,9 @@ synchronous or asynchronous IO. The `msync()` calls are made for column files
 only, so while the `sync`/`async` commit modes improve the overall durability,
 they don't guarantee durability in the face of OS errors or power loss.
 
+From QuestDB 7.2 onwards, `fsync()` calls are used to improve durability over OS
```
Comment (puzpuzpuz) [inline docs/concept/storage-model.md:76]: We were using msync/fsync before 7.2. What 7.2 changes is that it adds a bunch of fsync/msync for both metadata and data files. Metadata files are sync always (most, but not all of them if we keep @bluestreak01 's PR as is) while data files are synced only when sync commit mode is configured by the user.

Comment (amyshwang) [reply docs/concept/storage-model.md]: I amended the wording - thank you.

File: docs/concept/storage-model.md [hunk 2]
```diff
@@ -73,6 +73,11 @@ synchronous or asynchronous IO. The `msync()` calls are made for column files
 only, so while the `sync`/`async` commit modes improve the overall durability,
 they don't guarantee durability in the face of OS errors or power loss.
 
+From QuestDB 7.2 onwards, data durability is improved over OS errors or power
```
Comment (puzpuzpuz) [inline docs/concept/storage-model.md:76]: I'd suggest removing this paragraph completely and change the previous one in the following way:

By default, QuestDB relies on OS-level data **durability** for data files leaving
the OS to write dirty pages to disk. Data durability can also be configured to invoke
`msync()`/`fsync()` on each commit at the cost of reduced ingestion throughput.
Consider enabling `sync` commit mode to improve data durability in the face of
OS errors or power loss:
```ini title=server.conf
cairo.commit.mode=sync
```

Comment (puzpuzpuz) [reply docs/concept/storage-model.md:76]: cc @bluestreak01 who might want to change the wording

Comment (amyshwang) [reply docs/concept/storage-model.md]: Updated using the suggestion!

File: docs/concept/storage-model.md [hunk 3]
```diff
@@ -64,14 +64,17 @@ does so lock-free to ensure minimal impact on concurrent reads.
...
-synchronous or asynchronous IO. The `msync()` calls are made for column files
-only, so while the `sync`/`async` commit modes improve the overall durability,
-they don't guarantee durability in the face of OS errors or power loss.
+By default, QuestDB relies on OS-level data **durability** for data files
+leaving the OS to write dirty pages to disk. Data durability can also be
+configured to invoke `msync()`/`fsync()` for column files on each commit at the
+cost of reduced ingestion throughput. Consider enabling the `sync` commit mode
+to improve data durability in the face of OS errors or power loss:
+
+```txt
```
Comment (puzpuzpuz) [inline docs/concept/storage-model.md:75]:
```suggestion
```ini title="server.conf"
```

**Pull request merged**